cookbook/controller/service.rst

```diff
@@ -232,6 +234,28 @@ inject *only* the exact service(s) that you need directly into the controller.
    are valid, exactly how you want to organize your reusable code is up to
    you.
 
+Base Controller Methods and their service replacements
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This table provides an overview how to do the convenience methods of the base
+controller yourself.
+
+=========================== ================================== ==========================================================================================================================================
+Method                      Service                            PHP Code
+=========================== ================================== ==========================================================================================================================================
+``createForm``              ``form.factory``                   ``$formFactory->create($type, $data, $options)``
+``createFormBuilder``       ``form.factory``                   ``$formFactory->createBuilder('form', $data, $options)``
+``createNotFoundException`` -                                  ``throw new NotFoundHttpException($message, $previous);``
+``forward``                 ``http_kernel``                    ``$httpKernel->forward($controller, $path, $query)``
+``generateUrl``             ``router``                         ``$router->generate($route, $params, $absolute)``
+``getDoctrine``             ``doctrine``
+``getRequest``              -                                  Use ``Request $request`` as parameter in your action method. The order of the parameters is not important, but the typehint must be there.
```

Comment:
@WouterJ one of these days we should standardize the table syntax to use in the documentation. Hopefully, we'll never recommend to use the rigid grid syntax ;)